internal/commands/buildpack_package.go

```diff
@@ -134,22 +138,33 @@ func BuildpackPackage(logger logging.Logger, cfg config.Config, packager Buildpa
 	cmd.Flags().BoolVar(&flags.Flatten, "flatten", false, "Flatten the buildpack into a single layer")
 	cmd.Flags().StringSliceVarP(&flags.FlattenExclude, "flatten-exclude", "e", nil, "Buildpacks to exclude from flattening, in the form of '<buildpack-id>@<buildpack-version>'")
 	cmd.Flags().IntVar(&flags.Depth, "depth", -1, "Max depth to flatten.\nOmission of this flag or values < 0 will flatten the entire tree.")
+	if !cfg.Experimental {
+		cmd.Flags().MarkHidden("flatten")
+		cmd.Flags().MarkHidden("depth")
+		cmd.Flags().MarkHidden("flatten-exclude")
+	}
 	AddHelpFlag(cmd, "package")
 	return cmd
 }
 
-func validateBuildpackPackageFlags(p *BuildpackPackageFlags) error {
+func validateBuildpackPackageFlags(cfg config.Config, p *BuildpackPackageFlags) error {
 	if p.Publish && p.Policy == image.PullNever.String() {
 		return errors.Errorf("--publish and --pull-policy never cannot be used together. The --publish flag requires the use of remote images.")
 	}
 	if p.PackageTomlPath != "" && p.Path != "" {
 		return errors.Errorf("--config and --path cannot be used together. Please specify the relative path to the Buildpack directory in the package config file.")
 	}
 
-	if p.Flatten && len(p.FlattenExclude) > 0 {
-		for _, exclude := range p.FlattenExclude {
-			if strings.Count(exclude, "@") != 1 {
-				return errors.Errorf("invalid format %s; please use '<buildpack-id>@<buildpack-version>' to exclude buildpack from flattening", exclude)
+	if p.Flatten {
+		if !cfg.Experimental {
+			return client.NewExperimentError("Flattening a buildpack package is currently experimental.")
```

Comment:
Maybe add in how they can enable experimental features, for users who don't know how

Reply:
We already have a message set [here](https://github.com/buildpacks/pack/blob/52902b093e04f3c37479def536cfaa54cc24ce17/internal/commands/commands.go#L77)